File: 8/8.js
// 시간 복잡도: O(n^m)
// 공간 복잡도: O(m)
function main(n, m) {
  const numbers = Array.from({ length: n }, (_, i) => i + 1);

  function dfs(level, result) {
    if (level === m) {
      console.log(result);
      return;
    }

    for (let i = 0; i < n; i++) {
      result.push(numbers[i]);
      dfs(level + 1, result);
      result.pop();
    }
  }

  dfs(0, []);
}

console.log(main(3, 2));

File: 8/8.py
def main(n, m):
    numbers: list[int] = list(range(1, n + 1))

    def dfs(level, result):
        if level == m:
            print(result)
            return
        for i in range(n):
            dfs(level + 1, result + [numbers[i]])

    dfs(0, [])


print(main(3, 2))
